linera-views/src/set_view.rs

```diff
@@ -888,3 +840,31 @@ where
         self.set.hash().await
     }
 }
+
+impl<C> DeleteStorageFirst for ByteSetView<C> {
+    fn delete_storage_first(&self) -> bool {
+        self.delete_storage_first
+    }
+}
+
+impl<C, I> DeleteStorageFirst for SetView<C, I> {
+    fn delete_storage_first(&self) -> bool {
+        self.set.delete_storage_first
+    }
+}
+
+impl<C, I> DeleteStorageFirst for CustomSetView<C, I> {
+    fn delete_storage_first(&self) -> bool {
+        self.set.delete_storage_first
+    }
+}
+
+/// Type wrapping `ByteSetView` while memoizing the hash.
+pub type MemoizedByteSetView<C> = WrappedHashableContainerView<C, ByteSetView<C>, HasherOutput>;
+
+/// Type wrapping `SetView` while memoizing the hash.
+pub type MemoizedSetView<C, I> = WrappedHashableContainerView<C, SetView<C, I>, HasherOutput>;
```

Comment:
`MemoizedXXXView` is not amazing. Perhaps `HashedXXXView` is ok.

We could also not define the type aliases. Especially, `WrappedHashableContainerView` is renamed to something shorter like `Hashed(Container)View`.

Reply:
Ok, done with `HashedXXXView`.